main.go

```diff
@@ -0,0 +1,112 @@
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+var (
+	command        string
+	outputFileName string
+)
+
+func main() {
+
+	checkArgs()
+
+	flag.Parse()
+
+	outputFile, err := getWriteFile(outputFileName)
+	if err != nil {
+		quit(1, err)
+	}
+
+	defer func() {
+		err = outputFile.Close()
+		if err != nil {
+			log.Fatal(err)
+		}
+	}()
+
+	err = execCommand(command, outputFile)
+	if err != nil {
+		quit(1, err)
+	}
+
+}
+
+func init() {
+	flag.StringVar(&command, "c", "", "Command to execute")
+	flag.StringVar(&outputFileName, "f", "outputFile", "Otput File")
+}
+
+func checkArgs() {
+	if len(os.Args) < 2 {
+		printUsage()
+		quit(0, nil)
+	}
+}
+
+func printUsage() {
+	fmt.Println("\nCommand Executor")
+	fmt.Println()
+	fmt.Printf("Usage: %s [option] [command or filename]", os.Args[0])
+	fmt.Println("\n\nOptions:")
+	flag.PrintDefaults()
+}
+
+func getWriteFile(fileName string) (*os.File, error) {
+	return os.OpenFile(fileName, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0600)
+}
+
+func parseCommand(command string) (string, []string) {
+	commandSlice := strings.Split(command, " ")
+	name := commandSlice[0]
+	args := []string{}
+	if len(commandSlice) > 1 {
+		args = commandSlice[1:]
+	}
+	return name, args
+}
+
+func execCommand(command string, outFile io.Writer) (err error) {
```

Comment:
Naming would not be specific, `outFile io.Writer` would be called just `writer` or `w`